File: frontend/src/objects/merchant/menu/MenuItemSelectionOption.ts
// Business note: frontend mirror of a backend service object; keep names and fields aligned unless this file is explicitly frontend-only.
import type { CurrencyCents, DisplayText } from '@/objects/core/SharedObjects'

export type MenuItemSelectionOption = {
  name: DisplayText
  additionalPriceCents: CurrencyCents
}

File: backend/src/main/scala/services/merchant/objects/MenuItemSelectionOption.scala
package services.merchant.objects

// Business note: service-owned domain object; mirror it in the frontend only when it is part of the protocol surface.
import system.objects.given

import system.objects.*
import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.*

final case class MenuItemSelectionOption(
    name: DisplayText,
    additionalPriceCents: CurrencyCents,
)
object MenuItemSelectionOption:
  given Encoder[MenuItemSelectionOption] = deriveEncoder
  given Decoder[MenuItemSelectionOption] = deriveDecoder
